File: agent/src/schemas/InvoiceRecord.py
from pydantic import BaseModel, Field


class InvoiceRecord(BaseModel):
    invoice_date: str = Field(alias="invoiceDate")
    date_work_completed: str = Field(alias="dateWorkCompleted")
    payment_due: str = Field(alias="paymentDue")
    estimate_reference: str = Field(alias="estimateReference")
    invoice_number: str = Field(alias="invoiceNumber")
    client: str
    property: str
    project_description: str = Field(alias="projectDescription")
    cost_to_client: str = Field(alias="costToClient")
    labor_expense: str = Field(alias="laborExpense")
    equipment_expense: str = Field(alias="equipmentExpense")
    materials_expense: str = Field(alias="materialsExpense")
    administrative_notes: str = Field(alias="administrativeNotes")
    completion_status: str = Field(alias="completionStatus")
    service_categories: list[str] = Field(alias="serviceCategories")

    model_config = {
        "populate_by_name": True,
    }

File: server/src/main/java/org/example/server/dto/InvoiceRecord.java
package org.example.server.dto;

import java.util.List;

/**
 * Mirrors the client's {@code InvoiceRecord} type
 * (client/app/hooks/useInvoices.ts). Field order matches the client exactly.
 *
 * <p>NOTE: {@code email} is intentionally absent — it is an email-only / non-sheet field on the
 * client and must never be read from or returned by this API.
 *
 * <p>{@code id} is the stable primary key, read from column A. It is server-owned (generated on
 * append) and used to locate the physical row for single-cell updates.
 */
public record InvoiceRecord(
        String id,
        String invoiceDate,
        String dateWorkCompleted,
        String paymentDue,
        String paymentStatus,
        String estimateReference,
        String invoiceNumber,
        String client,
        String property,
        String projectDescription,
        String costToClient,
        String laborExpense,
        String equipmentExpense,
        String materialsExpense,
        String administrativeNotes,
        List<String> serviceCategories) {
}
